tests/components/esphome/test_config_flow.py

```diff
@@ -218,3 +218,67 @@ def __init__(self):
     assert result['errors'] == {
         'base': 'connection_error'
     }
+
+
+async def test_discovery_initiation(hass, mock_client):
+    """Test discovery importing works."""
+    flow = config_flow.EsphomeFlowHandler()
+    flow.hass = hass
+    service_info = {
+        'host': '192.168.43.183',
+        'port': 6053,
+        'hostname': 'test8266.local.',
+        'properties': {}
+    }
+    result = await flow.async_step_discovery(user_input=service_info)
+    assert result['type'] == 'form'
+    assert result['step_id'] == 'user'
+
+    # Validate schema values
+    for key in result['data_schema'].schema:
```

Comment:
What are you doing here?

Reply:
Why not just give the user the option to adopt it and that's it? Why would they need to look at the info ?

Reply:
Validating the schema value defaults are correct. I know it's messy but the best I could come up with.

Also tried things like
```yaml
assert result['data_schema'] == {
  vol.Optional('host', default='test8266.local'): str,
  vol.Optional('port', default= 6053): int,
}
```

but then I tried changing the default, and it still passed 🧐

Reply:
> Why not just give the user the option to adopt it and that's it? Why would they need to look at the info ?

Hmm, good idea!